### ISSUE TITLE
[NEW ALGORITHM] Next Permutation

### ISSUE THREAD

**Issue by Dipanita45** [AlgoGenesis/C#1048]
---

 Name:  
```[Next Permutation]```  

 About:  
The next_permutation function is a part of the C++ Standard Library that rearranges the elements in the range [first, last) into the next lexicographically greater permutation. In simpler terms, it takes a sequence of elements and transforms it into a sequence that would appear later if all possible sequences were sorted lexicographically. If the function can find a next permutation, it returns true; otherwise, it transforms the range into the lexicographically first permutation and returns false.
---

### Labels:  
```new algorithm, gssoc-ext, hacktoberfest, level1```

---

### Assignees:  
- [yes] Contributor in GSSoC-ext
- [yes] Want to work on it

Please assign me on the issue @pankaj-bind

**Comment by Dipanita45** [comment]
Please assign me the issue 
@pankaj-bind

**Comment by pankaj-bind** [comment]
@Dipanita45 assigned
